float32 centre_x
float32 centre_y
float32 centre_z
int32 index
float32 radius

